lib/hardware_services/inventory_drivers/QuadsNativeInventoryDriver.py

```diff
@@ -153,12 +153,45 @@ def load_data(self, quadsinstance, force):
             quadsinstance.logger.error(ex)
             exit(1)
 
-    def write_data(self, quadsinstance, doexit = True):
-        quadsinstance.quads_write_data(doexit)
+    def write_data(self, quadsinstance, doexit):
+        try:
+            stream = open(quadsinstance.config, 'w')
+            quadsinstance.data = {"clouds":quadsinstance.quads.clouds.data,
+             "hosts":quadsinstance.quads.hosts.data, "history":quadsinstance.quads.history.data, "cloud_history":quadsinstance.quads.cloud_history.data}
+            stream.write( yaml.dump(quadsinstance.data, default_flow_style=False))
+            if doexit:
+                exit(0)
+        except Exception, ex:
+            quadsinstance.logger.error("There was a problem with your file %s" % ex)
```

Comment:
For when calling `.error`, it is usually the case that the message will be logged, but when using other methods, like `.info` and `.debug`, the string formatting will be invoked to construct the log message, but if that logging level is not being emitted, then all the formatting time is a waste.  Best to let the logger format the string when it knows the message will be emitted.